service/discovery/azure/compute.go

```diff
@@ -136,6 +144,78 @@ func (d *azureComputeDiscovery) List() (list []voc.IsCloudResource, err error) {
 	return
 }
 
+// Discover functions and web apps
+func (d *azureComputeDiscovery) discoverVirtualMachineScaleSets() ([]voc.IsCloudResource, error) {
+	var list []voc.IsCloudResource
+
+	// initialize virtual machine scale set client
+	if err := d.initVirtualMachineScaleSetClient(); err != nil {
+		return nil, err
+	}
+
+	// List all VMs
+	err := listPager(d.azureDiscovery,
+		d.clients.virtualMachineScaleSetClient.NewListAllPager,
+		d.clients.virtualMachineScaleSetClient.NewListPager,
+		func(res armcompute.VirtualMachineScaleSetsClientListAllResponse) []*armcompute.VirtualMachineScaleSet {
+			return res.Value
+		},
+		func(res armcompute.VirtualMachineScaleSetsClientListResponse) []*armcompute.VirtualMachineScaleSet {
+			return res.Value
+		},
+
+		func(scaleSet *armcompute.VirtualMachineScaleSet) error {
+			r, err := d.discoverVMScaleSetVMs(scaleSet)
```

Comment:
Shouldn't that be `handleXXX` to be consistent?

Reply:
Ok, it seems to be a nested discover-then-handle-function

Reply:
No, because first the VMs in the scale set must be discovered and then the handleX method is called.